src/components/input/input.spec.ts

```diff
@@ -0,0 +1,180 @@
+import {
+  fakeAsync,
+  inject,
+  ComponentFixture,
+  TestComponentBuilder,
+  injectAsync,
+  tick,
+} from 'angular2/testing';
+
+import {
+  it,
+  expect,
+  beforeEach,
+} from '../../core/facade/testing';
+
+import {Component} from 'angular2/core';
+import {By} from 'angular2/platform/browser';
+
+import {
+  MdInput, MdInputDuplicatedHintException, MD_INPUT_DIRECTIVES,
+  MdInputPlaceholderConflictException, MdInputUnsupportedTypeException
+} from './input';
+
+
+export function main() {
+  describe('MdInput', function () {
+    var builder: TestComponentBuilder;
+
+    beforeEach(inject([TestComponentBuilder], function (tcb: TestComponentBuilder) {
+      builder = tcb;
+    }));
+
+    it('creates a native <input> element', injectAsync([], () => {
+      return builder.createAsync(MdInputBaseTestController)
+        .then((fixture) => {
+          fixture.detectChanges();
+          expect(fixture.debugElement.query(By.css('input'))).not.toBeNull();
+        });
+    }));
+
+    it('implements ControlValueAccessor', injectAsync([], () => {
+      return builder.createAsync(MdInputBaseTestController)
+        .then((fixture) => {
+          fixture.detectChanges();
+          fakeAsync(() => {
+            let instance = fixture.componentInstance;
+            let component = fixture.debugElement.query(By.directive(MdInput)).componentInstance;
+            let el: HTMLInputElement = fixture.debugElement.query(By.css('input')).nativeElement;
+
+            instance.model = 'hello';
+            fixture.detectChanges();
+            tick();
+            expect(el.value).toEqual('hello');
+
+            component.value = 'world';
+            fixture.detectChanges();
+            tick();
+            expect(el.value).toEqual('world');
+          })();
+        });
+    }));
+
+    it('counts characters', injectAsync([], () => {
+      return builder.createAsync(MdInputBaseTestController)
+        .then((fixture) => {
+          let instance = fixture.componentInstance;
+          fixture.detectChanges();
+          let inputInstance = fixture.debugElement.query(By.directive(MdInput)).componentInstance;
+          expect(inputInstance.characterCount).toEqual(0);
+
+          instance.model = 'hello';
+          fixture.detectChanges();
+          expect(inputInstance.characterCount).toEqual(5);
+        });
+    }));
```

Comment:
There's no standard for character count, so really it's two tests; one that check the property is what's expected, and the other that checks the hint label is what's expected. The character count is simply a usage of both.